src/App.tsx

```diff
@@ -1,14 +1,47 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [search, setSearch] = useState<string>('');
+  const [status, setStatus] = useState<Status>(Status.All);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Try Again Later'))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filteredTodos = todos
+    .filter((todo) => {
+      switch (status) {
+        case Status.All:
+          return true;
+        case Status.Active:
+          return !todo.completed;
+        case Status.Completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    })
+    .filter((todo) => todo.title.toLowerCase().includes(search.toLowerCase()));
```

Comment:
```suggestion
  const filteredTodos = todos.filter(todo => {
      const includesQuery = todo.title.toLocaleLowerCase()
        .includes(query.toLocaleLowerCase());

      switch (status) {
        case Status.Active:
          return !todo.completed && includesQuery;

        case Status.Completed:
          return todo.completed && includesQuery;

        case Status.All:
        default:
          return includesQuery;
      }
    });
```

No need to use `filter` twice